src/chains/ethereum_1/templates/chainActions/erc20_approve.ts

```diff
@@ -0,0 +1,50 @@
+// import { toHex } from 'web3-utils'
+import {
+  EthereumAddress,
+  EthereumTransactionAction,
+  EthereumChainActionType,
+  EthereumDecomposeReturn,
+} from '../../models'
+import { erc20Abi } from '../abis/erc20Abi'
+import { toEthereumAddress, ethereumTrxArgIsNullOrEmpty } from '../../helpers'
+import { getArrayIndexOrNull } from '../../../../helpers'
+
+interface erc20ApproveParams {
+  contractAddress: EthereumAddress
+  from?: EthereumAddress
+  spender: EthereumAddress
+  value: number
+}
+
+export const composeAction = ({ contractAddress, from, spender, value }: erc20ApproveParams) => {
+  const contract = {
+    abi: erc20Abi,
+    parameters: [spender, value],
+    method: 'approve',
+  }
+  return {
+    to: contractAddress,
+    from,
+    contract,
+  }
+}
+
+export const decomposeAction = (action: EthereumTransactionAction): EthereumDecomposeReturn => {
+  const { to, from, contract } = action
+  if (contract?.abi === erc20Abi && contract?.method === 'approve') {
+    const returnData: Partial<erc20ApproveParams> = {
+      contractAddress: to,
+      from,
+      spender: toEthereumAddress(getArrayIndexOrNull(contract?.parameters, 0) as string),
+      value: getArrayIndexOrNull(contract?.parameters, 1) as number,
+    }
+    const partial = !returnData?.from || ethereumTrxArgIsNullOrEmpty(to)
```

Comment:
As I mentioned in another PR comments. !value wont work if the value can be 0 or false - since !value would return true in these cases. You need to use !isNullOrEmpty everywhere you do partial = ...
Please make sure you read my comments in the PR and then replace all these in ETh and EOS decompose functions